pkg/api/aim/query/query.go

```diff
@@ -613,53 +589,84 @@ func (pq *parsedQuery) parseName(node *ast.Name) (any, error) {
 				},
 			), nil
 		case "re":
-			return callable(
-				func(args []ast.Expr) (any, error) {
-					operation, ok := args[0].(*ast.Str)
-					if !ok {
-						return nil, errors.New("re operation type has to be ast.Str")
-					}
-					switch operation.S {
-					case RegexpOperationMatch:
-						fallthrough
-					case RegexpOperationSearch:
-						// subtract 1, because first argument is an operation argument.
-						if len(args)-1 != 2 {
-							return nil, errors.New("re.match function support exactly 2 arguments")
-						}
+			return attributeGetter(
+				func(attr string) (any, error) {
+					switch attr {
+					case "match":
+						return callable(
+							func(args []ast.Expr) (any, error) {
+								if len(args) != 2 {
+									return nil, errors.New("re.match function support exactly 2 arguments")
+								}
 
-						regexp, ok := args[1].(*ast.Str)
-						if !ok {
-							return nil, errors.New("first argument type for re.match function has to be ast.Str")
-						}
-						attribute, ok := args[2].(*ast.Attribute)
-						if !ok {
-							return nil, errors.New("second argument type for re.match function has to be ast.Attribute")
-						}
+								parsedNode, err := pq.parseNode(args[0])
+								if err != nil {
+									return nil, err
+								}
+								str, ok := parsedNode.(string)
+								if !ok {
+									return nil, errors.New("first argument type for re.match function has to be ast.Str")
+								}
 
-						attributes, err := pq.parseName(attribute.Value.(*ast.Name))
-						if err != nil {
-							return nil, err
-						}
+								parsedNode, err = pq.parseNode(args[1])
+								if err != nil {
+									return nil, err
+								}
 
-						column, err := attributes.(attributeGetter)(string(attribute.Attr))
-						if err != nil {
-							return nil, err
-						}
+								column, ok := parsedNode.(clause.Column)
+								if !ok {
+									return nil, errors.New(
+										"second argument type for re.match function has to be clause.Column",
+									)
+								}
 
-						// handle difference between `re.match` and `re.search`.
-						if operation.S == RegexpOperationMatch {
-							regexp.S = py.String(fmt.Sprintf("^%s", regexp.S))
-						}
-						return Regexp{
-							Eq: clause.Eq{
-								Column: column,
-								Value:  regexp.S,
+								return Regexp{
+									Eq: clause.Eq{
+										Column: column,
+										Value:  fmt.Sprintf("^%s", str),
+									},
+									Dialector: pq.qp.Dialector,
+								}, nil
 							},
-							Dialector: pq.qp.Dialector,
-						}, nil
+						), nil
+					case "search":
+						return callable(
+							func(args []ast.Expr) (any, error) {
```

Comment:
could you please make `match` and `search` be the same except for the `^` character?

Reply:
probably I can and it was before but I decided to just keep since a bit more clean even if they are almost the same. can we keep them split?

Reply:
no, I would really like us to stick to the DRY principle — maybe fold them into one private method of `parsedQuery`?

Reply:
done.